phoenix-core-client/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -1788,6 +1816,10 @@ public static void annotateMutationWithConditionalTTL(
         byte[] ttl = ttlExpr.serialize();
         for (Mutation mutation : mutations) {
             mutation.setAttribute(BaseScannerRegionObserverConstants.TTL, ttl);
+            if (!table.isStrictTTL()) {
+                mutation.setAttribute(BaseScannerRegionObserverConstants.IS_STRICT_TTL,
+                        PBoolean.INSTANCE.toBytes(table.isStrictTTL()));
+            }
```

Comment:
This is improvement: we don't need to send the mutation and scan attribute `_IS_STRICT_TTL` if the TTL is already strict.